src/runtimes/runners/TaskCommandRunnerFactory.ts

```diff
@@ -0,0 +1,79 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as os from 'os';
+import * as vscode from 'vscode';
+import { CommandResponse, CommandResponseLike, CommandRunner, ICommandRunnerFactory, normalizeCommandResponseLike } from '@microsoft/container-runtimes';
+
+interface TaskCommandRunnerOptions {
+    taskName: string;
+    workspaceFolder?: vscode.WorkspaceFolder;
+    cwd?: string;
+    alwaysRunNew?: boolean;
+    rejectOnError?: boolean;
+    focus?: boolean;
+    env?: never; // Environment is not needed and should not be used, because VSCode adds it already (due to using `ExtensionContext.environmentVariableCollection`)
+}
+
+export class TaskCommandRunnerFactory implements ICommandRunnerFactory {
+    public constructor(private readonly options: TaskCommandRunnerOptions) {
+    }
+
+    public getCommandRunner(): CommandRunner {
+        return async <T>(commandResponseLike: CommandResponseLike<T>) => {
+            const commandResponse: CommandResponse<T> = await normalizeCommandResponseLike(commandResponseLike);
+            await executeAsTask(this.options, commandResponse.command, commandResponse.args);
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            return undefined!;
+        };
+    }
+}
+
+async function executeAsTask(options: TaskCommandRunnerOptions, command: string, args?: vscode.ShellQuotedString[]): Promise<void> {
+    const shellExecutionOptions = { cwd: options.cwd || options.workspaceFolder?.uri?.fsPath || os.homedir() };

```

Comment:
Do you also want to allow for passing additional environment variables?

Reply:
We can probably put off adding that until we need it. The environment variables in the `containers.environment` setting will be automatically applied by VSCode itself.

Reply:
sounds good